modules/postfix.lua

```diff
@@ -0,0 +1,524 @@
+                                  * l.Cg(l.P(1)^1, 'postfix_scache_timestamp')
+
+-- aggregate all patterns
+local postfix_patterns = {
+  smtpd = l.Ct(postfix_smtpd_connect_cg
+        + postfix_smtpd_disconnect_cg
+        + postfix_smtpd_lostconn_cg
+        + postfix_smtpd_noqueue_cg
+        + postfix_smtpd_pipelining_cg
+        + postfix_tlsconn_cg
+        + postfix_warning_cg
+        + postfix_smtpd_proxy_cg
+        + postfix_keyvalue_greedy_cg),
+  cleanup =  l.Ct(postfix_cleanup_milter_cg
+          + postfix_warning_cg
+          + postfix_keyvalue_greedy_cg),
+  qmgr = l.Ct(postfix_qmgr_removed_cg
+       + postfix_qmgr_active_cg
+       + postfix_warning_cg),
+  pipe = l.Ct(postfix_pipe_delivered_cg
+       + postfix_pipe_forward_cg),
+  postscreen = l.Ct(postfix_ps_connect_cg
+             + postfix_ps_access_cg
+             + postfix_ps_noqueue_cg
+             + postfix_ps_toobusy_cg
+             + postfix_ps_cache_cg
+             + postfix_ps_dnsbl_cg
+             + postfix_ps_violations_cg
+             + postfix_warning_cg),
+  dnsblog = l.Ct(postfix_dnsblog_listing_cg),
+  anvil = l.Ct(postfix_anvil_conn_rate_cg
+        + postfix_anvil_conn_cache_cg
+        + postfix_anvil_conn_count_cg),
+  smtp = l.Ct(postfix_smtp_delivery_cg
+       + postfix_smtp_connerr_cg
+       + postfix_smtp_lostconn_cg
+       + postfix_tlsconn_cg
+       + postfix_warning_cg),
+  discard = l.Ct(postfix_queueid_cg
+          * l.P': '
+          * l.Cg((l.P(1) - l.P' status=')^1, 'postfix_keyvalue_data')
+          * l.P' status='
+          * l.Cg(l.alnum^1, 'postfix_status')),
+  -- lmtp = smtp,
+  pickup = l.Ct(postfix_queueid_cg
+         * l.P': uid='
+         * l.Cg((l.P(1) - l.P' from=')^1, 'postfix_uid')
+         * l.P' from=<'
+         * l.Cg((l.P(1) - l.P'>')^1, 'postfix_from')
+         * (l.P'> orig_id='
+           * l.Cg(l.P(1)^1, 'postfix_orig_id'))^-1
+         ),
+  tlsproxy = l.Ct(postfix_tlsproxy_conn_cg),
+  master = l.Ct(postfix_master_start_cg
+         + postfix_master_exit_cg),
+  bounce = l.Ct(postfix_bounce_notification_cg),
+  sendmail = l.Ct(postfix_warning_cg),
+  postdrop = l.Ct(postfix_warning_cg),
+  scache = l.Ct(postfix_scache_lookups_cg
+         + postfix_scache_simultaneous_cg
+         + postfix_scache_timestamp_cg),
+  trivial_rewrite = l.Ct(postfix_warning_cg),
+  tlsmgr = l.Ct(postfix_warning_cg),
+  ['local'] = l.Ct(postfix_keyvalue_greedy_cg),
+}
+postfix_patterns['lmtp'] = postfix_patterns['smtp']
+
+--local name = l.C(l.alpha^1) * space
+--local value = l.C((l.P(1) - l.P',')^1)
+--local sep = l.S(',;') * space
+--local pair = l.Cg(name * '=' * space * value) * sep^-1
+--local queueid = lpeg.Cf(lpeg.Ct'' * l.Cg(lpeg.Cc'queueid' * l.C(l.xdigit ^ 1)) * l.P': ', rawset)
+--local postfix_grammar = l.Cf(queueid * pair^0, rawset)
+
+
+
+local postfix_programname = l.P'postfix'
+                          * (l.P(1) - l.P'/')^0
+                          * '/'
+                          * l.Cg(l.P(1)^1)
```

Comment:
fyi: `l.C` would suffice the anonymous capture group buys you nothing here (but it doesn't hurt either)